server/service/src/vaccination/update/generate.rs

```diff
@@ -62,44 +64,53 @@ pub fn generate(
 
     let update_transactions = update_input.update_transactions.unwrap_or(false);
 
+    let should_revert = new_stock_line.is_none() && existing_prescription_line.is_some();
+
     // Reverse prescription if it existed, and the stock line is changing
-    let create_customer_return = if stock_line_has_changed && update_transactions {
-        existing_stock_line.map(|stock_line| {
-            let amount = get_dose_as_number_of_packs(&stock_line);
-            let stock_line_row = stock_line.stock_line_row;
-
-            let create_return = InsertCustomerReturn {
-                id: uuid(),
-                other_party_id: patient_id.clone(),
-                is_patient_return: true,
-                outbound_shipment_id: None,
-                customer_return_lines: vec![CustomerReturnLineInput {
+    let create_customer_return = if (stock_line_has_changed || should_revert) && update_transactions
+    {
+        existing_prescription_line
```

Comment:
added in prescription line - realised there were a couple MORE cases:

- If you give (and create prescription) -> change batch and _don't_ update transactions -> un-give and _do_ revert transactions, it was creating a return of the new batch, rather than what was in the prescription!
- Also needed to handle give (create prescription) -> change to Other facility (which clears vaccination `stock_line`) but don't revert prescription -> reopen modal and click revert... so needed to check if the prescrizzle existed rather than if the vaccination stock line was changing!

Reply:
can use and_then instead of map.flatten (: